app/app.go

```diff
@@ -112,6 +116,52 @@ func (a *app) CreateIntegration(
 	return result, err
 }
 
+func (a *app) SetIntegrationCredentials(
+	ctx context.Context,
+	integrationID uuid.UUID,
+	credentials model.Credentials,
+) error {
+	err := a.store.SetIntegrationCredentials(ctx, integrationID, credentials)
+	if err != nil {
+		switch cause := errors.Cause(err); cause {
+		case store.ErrObjectNotFound:
+			return ErrIntegrationNotFound
+		default:
+			return err
+		}
+	}
+	return err
+}
+
+func (a *app) RemoveIntegration(
+	ctx context.Context,
+	integrationID uuid.UUID,
+) error {
+	// check if there are any devices with given integration enabled
+	devices, err := a.store.GetDevicesByIntegrationID(ctx, integrationID)
+	if err != nil {
+		switch cause := errors.Cause(err); cause {
+		case store.ErrObjectNotFound:
+			return ErrDevicesNotFound
```

Comment:
I don't get how you can get `ErrObjectNotFound` from the store layer when calling `GetDevicesByIntegrationID`.
What are you testing here, exactly?

Reply:
Fixed, thanks!